src/main/java/Deadline.java

```diff
@@ -0,0 +1,34 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task {
+    protected LocalDateTime by;
+
+    public Deadline(String description, String by) {
+        super(description, TaskType.DEADLINE);
+        this.by = parseDateTime(by);
+    }
+
+    private LocalDateTime parseDateTime(String dateTime) {
```

Comment:
I like that even within the method, the code is broken up in chunks so that it is easier to read.